lib/agenda/index.js

```diff
@@ -33,17 +33,29 @@ class Agenda extends Emitter {
 
     this._isLockingOnTheFly = false;
     this._jobsToLock = [];
+    this._isReady = false;
```

Comment:
Won't this need a getter to check when `agenda.mongo` or `agenda.db` is set to update `agenda._isReady`?

Reply:
Below here the only time callback is run is when mongo or db is passed so if they're not passed it'll never be ready even if the fields are updated after initialization.